.circleci/config.yml

```diff
@@ -76,23 +113,55 @@ jobs:
       - run:
           name: Lint Client and Server
           command: |
-            ./node_modules/.bin/eslint-eslines $(git diff --name-only --diff-filter=d origin/master... | grep -E '^(client|server)' | grep -E '\.jsx?$' | circleci tests split)
+            # We'll use failure code to do some additional processing and report failure manually
+            set +o errexit
+
+            # Run the lint
+            npx --no-install                                          \
+              eslint-eslines                                          \
+              $(                                                      \
+                git diff --name-only --diff-filter=d origin/master... \
+                | grep -E '^(client|server)'                          \
+                | grep -E '\.jsx?$'                                   \
+                | circleci tests split                                \
+              )                                                       \
+              -- --format junit                                       \
+              > ~/eslint-results.xml
+
+            # Ensure we don't upload empty reports
+            # Eslint would not produce a malformed empty xml using the --format and -o options.
+            # However, with eslint-eslines and the redirected > output, we may get an empty file.
+            # We only put the file in to the reports directory if the status code informs us that
+            # lint issues have been encountered.
+            ESLINT_RESULT="$?"
+            if [[ "$ESLINT_RESULT" != "0" ]]; then
```

Comment:
does this not work?

```bash
if [ $? -ne 0 ]; then
```

Reply:
The `[ $VAR -ne 0 ]` bit should work (thanks), but we need to store the exit status in another variable so we can run other commands then exit with the status we're interested in (eslint's) so CI reads it. I may not have expressed that bit well in the comment.